src/components/VideoPlayer/BaseVideoPlayer.js

```diff
@@ -156,45 +160,37 @@ function BaseVideoPlayer({
     }, [currentVideoPlayerRef, handleFullscreenUpdate, handlePlaybackStatusUpdate]);
 
     useEffect(() => {
-        if (!isUploading) {
+        if (!isUploading || !videoPlayerRef.current) {
             return;
         }
 
         // If we are uploading a new video, we want to immediately set the video player ref.
         currentVideoPlayerRef.current = videoPlayerRef.current;
     }, [url, currentVideoPlayerRef, isUploading]);
 
-    useEffect(() => {
-        shouldUseSharedVideoElementRef.current = shouldUseSharedVideoElement;
-    }, [shouldUseSharedVideoElement]);
-
-    useEffect(
-        () => () => {
-            if (shouldUseSharedVideoElementRef.current) {
-                return;
-            }
-
-            // If it's not a shared video player, clear the video player ref.
-            currentVideoPlayerRef.current = null;
-        },
-        [currentVideoPlayerRef],
-    );
-
     // update shared video elements
     useEffect(() => {
-        if (shouldUseSharedVideoElement || url !== currentlyPlayingURL) {
+        if (shouldUseSharedVideoElement || url !== currentlyPlayingURL || isFullScreenRef.current) {
```

Comment:
`isFullScreenRef` is responsible for locking the video player when screen orientation changes in full-screen mode. It prevents video players from changing their state because of device rotation or window size changes. Changing state causes the full screen to dismiss. We need it here to block updating currently shared video element

Reply:
Hi, @Skalakid 
I'm also curious about test steps that cause unexpected errors when we remove `isFullScreenRef.current`.

Reply:
Hi @tienifr @jacobnguyen0000 here is a video that shows why we need `isFullScreenRef.current` ref:

| Without ref  | With ref |
| ------------- | ------------- |
| <video src="https://github.com/Expensify/App/assets/39538890/68aec9fd-5d55-4eac-a9b4-595c53f3a7f2">  | <video src="https://github.com/Expensify/App/assets/39538890/d3e38dc0-984d-49af-866c-29be3c2740f9">|

Steps:
1. Open chat
2. Open video attachment
3. Enter full screen mode
4. Rotate device
5. Verify if video is still in full screen mode
6. Rotate device to the previous position
7. Verify video state like in step 5

Reply:
Are you trying to remove this variable globally or only in this `useEffect` if statement?

Reply:
@Skalakid 
Thank you for your confirm.
Could you please let me know if this issue is reproducible on native devices?

Reply:
Confirmed again, it's not reproducible on my end.

Reply:
@Skalakid 
thank you for your check.